### PR TITLE
pull-request-template: Add the link on tests documentation?

### PR DESCRIPTION
**Pull request by ardumont**
###### Motivation for this change

When i recently opened a PR [1], i realized i missed test thanks to this guide.

But reading the arborescence tests tree link from this guide overwhelmed me.
I was able somehow to replicate a test but did not know how to run it.
That's clearly well explained in the doc (which somehow i did not find myself...T.T) [2]

So why not link it from the guide for people willing to dig in?

[1] https://github.com/NixOS/nixpkgs/pull/57036

[2] thanks gchristensen and clever for the help on #nixos

Cheers,

### CODE REVIEW

**Comment by danbst** [comment]
I think there should be only one link - to tests manual. In the tests manual chapters "writing test" and "running tests" should be flipped. So the order is like this:
- how to run tests
- list of tests (non-exhaustive, just to show what can be tested - simple test, simple test with GRUB/UEFI, graphical subsystem, SQL database, network of machines with frontend and backend, declarative containers)
- how to debug tests interactively
- and finally, how to write tests

@ardumont what do you think, would that order of sections be helpful? This reordering can be done in https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/development/nixos-tests.xml and https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/development/development.xml

**Comment by ardumont** [comment]
> I think there should be only one link - to tests manual.

I thought so as well but i was not so blunt as to propose the removal of the initial one.
Because there might have been reasons for it...

So do i adapt the commit to drop the first link?

>  In the tests manual chapters "writing test" and "running tests" should be flipped. 

Possibly. I read the full chapter prior to act on it.
But indeed, as the `writing tests` chapter is quite long.
It could possibly inhibit users from reaching `how to run tests`.
Then again, there is a toc in the manual so i'm not sure if the flipping is that needed.

> list of tests (non-exhaustive, just to show what can be tested - simple test, simple test with GRUB/UEFI, graphical subsystem, SQL database, network of machines with frontend and backend, declarative containers)

That's a good idea ;)

**Comment by danbst** [comment]
> I thought so as well but i was not so blunt as to propose the removal of the initial one.

this section is for newcomers, so newcomer's opinion is valuable here!

> So do i adapt the commit to drop the first link?

Yup. Manual has that link in the first line anyway

**Comment by ardumont** [comment]
I had forgotten about this, updated!

Cheers,

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by ardumont** [comment]
ping?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/do
    - [ ] NixOS
    - [ ] macOS
    - [ ] other Linux distributions
-- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
+- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change ([cf. manual documentation on how to run/write/debug tests](https://nixos.org/nixos/manual/index.html#sec-nixos-tests))
```
Comment (SuperSandro2000) [inline .github/PULL_REQUEST_TEMPLATE.md:23]:
```suggestion
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change ([cf. manual documentation on how to run/write/debug tests](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests))
```

Comment (ardumont) [reply .github/PULL_REQUEST_TEMPLATE.md]: hello, thx, adapted.

**Comment by Artturin** [comment]
Thank your for your pr, a section similar to this Is now in the template

**Comment by ardumont** [comment]
Great, thanks for the heads up.

Cheers,

**Pull request closed without merging**